I've taken the reproducing code from #4950 and added it to `tests/test_widgets.py`:
```
def test_4950():
    with pymupdf.open() as document:
        page = document.new_page()
        page.set_rotation(90)

        # Add a widget with a zero-dimension rect (simulating an invisible signature field)
        widget = pymupdf.Widget()
        widget.field_type = pymupdf.PDF_WIDGET_TYPE_SIGNATURE
        widget.rect = pymupdf.Rect(0, 0, 0, 0)
        page.add_widget(widget)

        page.remove_rotation()  # raises ValueError: bad rect
```
Unfortunately it seems to fail when pymupdf is built from https://github.com/thakoreh/PyMuPDF.git branch `fix/remove-rotation-empty-widget`.

Here's the output from `pytest -k test_4950 -sv tests`:
```
[+34.0s]: venv-aptest-3.13.12-64: test: pymupdf: test pymupdf: __________________________________ test_4950 ___________________________________                                                                                                                           
[+34.0s]: venv-aptest-3.13.12-64: test: pymupdf: test pymupdf: Traceback (most recent call last):                                                                                                                                                                         
[+34.0s]: venv-aptest-3.13.12-64: test: pymupdf: test pymupdf:   File "/home/jules/artifex/venv-aptest-3.13.12-64/lib/python3.13/site-packages/_pytest/runner.py", line 353, in from_call                                                                                 
[+34.0s]: venv-aptest-3.13.12-64: test: pymupdf: test pymupdf:     result: TResult | None = func()                                                                                                                                                                        
[+34.0s]: venv-aptest-3.13.12-64: test: pymupdf: test pymupdf:                              ~~~~^^                                                                                                                                                                        
[+34.0s]: venv-aptest-3.13.12-64: test: pymupdf: test pymupdf:   File "/home/jules/artifex/venv-aptest-3.13.12-64/lib/python3.13/site-packages/_pytest/runner.py", line 245, in <lambda>                                                                                  
[+34.0s]: venv-aptest-3.13.12-64: test: pymupdf: test pymupdf:     lambda: runtest_hook(item=item, **kwds),                                                                                                                                                               
[+34.0s]: venv-aptest-3.13.12-64: test: pymupdf: test pymupdf:             ~~~~~~~~~~~~^^^^^^^^^^^^^^^^^^^                                                                                                                                                                
[+34.0s]: venv-aptest-3.13.12-64: test: pymupdf: test pymupdf:   File "/home/jules/artifex/venv-aptest-3.13.12-64/lib/python3.13/site-packages/pluggy/_hooks.py", line 512, in __call__                                                                                   
[+34.0s]: venv-aptest-3.13.12-64: test: pymupdf: test pymupdf:     return self._hookexec(self.name, self._hookimpls.copy(), kwargs, firstresult)                                                                                                                          
[+34.0s]: venv-aptest-3.13.12-64: test: pymupdf: test pymupdf:            ~~~~~~~~~~~~~~^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^                                                                                                                          
[+34.0s]: venv-aptest-3.13.12-64: test: pymupdf: test pymupdf:   File "/home/jules/artifex/venv-aptest-3.13.12-64/lib/python3.13/site-packages/pluggy/_manager.py", line 120, in _hookexec                                                                                
[+34.0s]: venv-aptest-3.13.12-64: test: pymupdf: test pymupdf:     return self._inner_hookexec(hook_name, methods, kwargs, firstresult)                                                                                                                                   
[+34.0s]: venv-aptest-3.13.12-64: test: pymupdf: test pymupdf:            ~~~~~~~~~~~~~~~~~~~~^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
[+34.0s]: venv-aptest-3.13.12-64: test: pymupdf: test pymupdf:   File "/home/jules/artifex/venv-aptest-3.13.12-64/lib/python3.13/site-packages/pluggy/_callers.py", line 167, in _multicall
[+34.0s]: venv-aptest-3.13.12-64: test: pymupdf: test pymupdf:     raise exception
[+34.0s]: venv-aptest-3.13.12-64: test: pymupdf: test pymupdf:   File "/home/jules/artifex/venv-aptest-3.13.12-64/lib/python3.13/site-packages/pluggy/_callers.py", line 139, in _multicall
[+34.0s]: venv-aptest-3.13.12-64: test: pymupdf: test pymupdf:     teardown.throw(exception)
[+34.0s]: venv-aptest-3.13.12-64: test: pymupdf: test pymupdf:     ~~~~~~~~~~~~~~^^^^^^^^^^^
[+34.0s]: venv-aptest-3.13.12-64: test: pymupdf: test pymupdf:   File "/home/jules/artifex/venv-aptest-3.13.12-64/lib/python3.13/site-packages/_pytest/logging.py", line 850, in pytest_runtest_call
[+34.0s]: venv-aptest-3.13.12-64: test: pymupdf: test pymupdf:     yield
[+34.0s]: venv-aptest-3.13.12-64: test: pymupdf: test pymupdf:   File "/home/jules/artifex/venv-aptest-3.13.12-64/lib/python3.13/site-packages/pluggy/_callers.py", line 139, in _multicall
[+34.0s]: venv-aptest-3.13.12-64: test: pymupdf: test pymupdf:     teardown.throw(exception)
[+34.0s]: venv-aptest-3.13.12-64: test: pymupdf: test pymupdf:     ~~~~~~~~~~~~~~^^^^^^^^^^^
[+34.0s]: venv-aptest-3.13.12-64: test: pymupdf: test pymupdf:   File "/home/jules/artifex/venv-aptest-3.13.12-64/lib/python3.13/site-packages/_pytest/capture.py", line 900, in pytest_runtest_call
[+34.0s]: venv-aptest-3.13.12-64: test: pymupdf: test pymupdf:     return (yield)
[+34.0s]: venv-aptest-3.13.12-64: test: pymupdf: test pymupdf:             ^^^^^
[+34.0s]: venv-aptest-3.13.12-64: test: pymupdf: test pymupdf:   File "/home/jules/artifex/venv-aptest-3.13.12-64/lib/python3.13/site-packages/pluggy/_callers.py", line 139, in _multicall
[+34.0s]: venv-aptest-3.13.12-64: test: pymupdf: test pymupdf:     teardown.throw(exception)
[+34.0s]: venv-aptest-3.13.12-64: test: pymupdf: test pymupdf:     ~~~~~~~~~~~~~~^^^^^^^^^^^
[+34.0s]: venv-aptest-3.13.12-64: test: pymupdf: test pymupdf:   File "/home/jules/artifex/venv-aptest-3.13.12-64/lib/python3.13/site-packages/_pytest/skipping.py", line 268, in pytest_runtest_call
[+34.0s]: venv-aptest-3.13.12-64: test: pymupdf: test pymupdf:     return (yield)
[+34.0s]: venv-aptest-3.13.12-64: test: pymupdf: test pymupdf:             ^^^^^
[+34.0s]: venv-aptest-3.13.12-64: test: pymupdf: test pymupdf:   File "/home/jules/artifex/venv-aptest-3.13.12-64/lib/python3.13/site-packages/pluggy/_callers.py", line 121, in _multicall
[+34.0s]: venv-aptest-3.13.12-64: test: pymupdf: test pymupdf:     res = hook_impl.function(*args)
[+34.0s]: venv-aptest-3.13.12-64: test: pymupdf: test pymupdf:   File "/home/jules/artifex/venv-aptest-3.13.12-64/lib/python3.13/site-packages/_pytest/runner.py", line 179, in pytest_runtest_call
[+34.0s]: venv-aptest-3.13.12-64: test: pymupdf: test pymupdf:     item.runtest()
[+34.0s]: venv-aptest-3.13.12-64: test: pymupdf: test pymupdf:     ~~~~~~~~~~~~^^
[+34.0s]: venv-aptest-3.13.12-64: test: pymupdf: test pymupdf:   File "/home/jules/artifex/venv-aptest-3.13.12-64/lib/python3.13/site-packages/_pytest/python.py", line 1720, in runtest
[+34.0s]: venv-aptest-3.13.12-64: test: pymupdf: test pymupdf:     self.ihook.pytest_pyfunc_call(pyfuncitem=self)
[+34.0s]: venv-aptest-3.13.12-64: test: pymupdf: test pymupdf:     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^^^^^^^^^^^^^^^^^
[+34.0s]: venv-aptest-3.13.12-64: test: pymupdf: test pymupdf:   File "/home/jules/artifex/venv-aptest-3.13.12-64/lib/python3.13/site-packages/pluggy/_hooks.py", line 512, in __call__
[+34.0s]: venv-aptest-3.13.12-64: test: pymupdf: test pymupdf:     return self._hookexec(self.name, self._hookimpls.copy(), kwargs, firstresult)
[+34.0s]: venv-aptest-3.13.12-64: test: pymupdf: test pymupdf:            ~~~~~~~~~~~~~~^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
[+34.0s]: venv-aptest-3.13.12-64: test: pymupdf: test pymupdf:   File "/home/jules/artifex/venv-aptest-3.13.12-64/lib/python3.13/site-packages/pluggy/_manager.py", line 120, in _hookexec
[+34.0s]: venv-aptest-3.13.12-64: test: pymupdf: test pymupdf:     return self._inner_hookexec(hook_name, methods, kwargs, firstresult)
[+34.0s]: venv-aptest-3.13.12-64: test: pymupdf: test pymupdf:            ~~~~~~~~~~~~~~~~~~~~^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
[+34.0s]: venv-aptest-3.13.12-64: test: pymupdf: test pymupdf:   File "/home/jules/artifex/venv-aptest-3.13.12-64/lib/python3.13/site-packages/pluggy/_callers.py", line 167, in _multicall
[+34.0s]: venv-aptest-3.13.12-64: test: pymupdf: test pymupdf:     raise exception
[+34.0s]: venv-aptest-3.13.12-64: test: pymupdf: test pymupdf:   File "/home/jules/artifex/venv-aptest-3.13.12-64/lib/python3.13/site-packages/pluggy/_callers.py", line 121, in _multicall
[+34.0s]: venv-aptest-3.13.12-64: test: pymupdf: test pymupdf:     res = hook_impl.function(*args)
[+34.0s]: venv-aptest-3.13.12-64: test: pymupdf: test pymupdf:   File "/home/jules/artifex/venv-aptest-3.13.12-64/lib/python3.13/site-packages/_pytest/python.py", line 166, in pytest_pyfunc_call
[+34.0s]: venv-aptest-3.13.12-64: test: pymupdf: test pymupdf:     result = testfunction(**testargs)
[+34.0s]: venv-aptest-3.13.12-64: test: pymupdf: test pymupdf:   File "/home/jules/artifex/PyMuPDF/tests/test_widgets.py", line 479, in test_4950
[+34.0s]: venv-aptest-3.13.12-64: test: pymupdf: test pymupdf:     page.add_widget(widget)
[+34.0s]: venv-aptest-3.13.12-64: test: pymupdf: test pymupdf:     ~~~~~~~~~~~~~~~^^^^^^^^
[+34.0s]: venv-aptest-3.13.12-64: test: pymupdf: test pymupdf:   File "/home/jules/artifex/venv-aptest-3.13.12-64/lib/python3.13/site-packages/pymupdf/__init__.py", line 10615, in add_widget
[+34.0s]: venv-aptest-3.13.12-64: test: pymupdf: test pymupdf:     widget._validate()
[+34.0s]: venv-aptest-3.13.12-64: test: pymupdf: test pymupdf:     ~~~~~~~~~~~~~~~~^^
[+34.0s]: venv-aptest-3.13.12-64: test: pymupdf: test pymupdf:   File "/home/jules/artifex/venv-aptest-3.13.12-64/lib/python3.13/site-packages/pymupdf/__init__.py", line 9121, in _validate
[+34.0s]: venv-aptest-3.13.12-64: test: pymupdf: test pymupdf:     raise ValueError("bad rect")
[+34.0s]: venv-aptest-3.13.12-64: test: pymupdf: test pymupdf: ValueError: bad rect
```

It's possible i've got myself confused, but perhaps you could add the test to your tree and include it in this PR, and we can then see whether it's a genuine problem when the pre-push test_quick is run?

Thanks.

--- a/src/__init__.py
+++ b/src/__init__.py
@@ -11933,6 +11933,8 @@ class Page:
                 pass
         for widget in self.widgets():  # modify field rectangles
             r = widget.rect * rot
+            if r.is_empty or r.is_infinite:
+                continue
             widget.rect = r
             widget.update()
         return rot  # the inverse of the generated derotation matrix

--- a/tests/test_widgets.py
+++ b/tests/test_widgets.py
@@ -465,3 +465,24 @@ def test_4114():
                 widget.update()
         document.save(path_out)
     assert values == expected_values2
+
+
+def test_4950():
+    with pymupdf.open() as document:
+        page = document.new_page()
+        page.set_rotation(90)
+
+        # Simulate an existing invisible signature field with a zero-dimension
+        # rectangle. Creating such a widget through add_widget() is rejected by
+        # validation, so create a valid widget first and then modify the PDF
+        # object directly.
+        widget = pymupdf.Widget()
+        widget.field_name = "Signature"
+        widget.field_type = pymupdf.PDF_WIDGET_TYPE_SIGNATURE
+        widget.rect = pymupdf.Rect(0, 0, 10, 10)
+        page.add_widget(widget)
+        document.xref_set_key(page.first_widget.xref, "Rect", "[0 0 0 0]")
+        page = document.reload_page(page)
+
+        page.remove_rotation()
+        assert page.rotation == 0

